### ISSUE TITLE
Error: Failed to create session or connect to servers.

### ISSUE THREAD

**Issue by F1reP0wer12345** [michaelherger/librespot#17]
I've recently been trying to setup Spotify inside of Kodi using [this](https://github.com/glk1001/plugin.audio.spotify) addon which mentions using spotty from you repo. When entering the credentials, it uses spotty with the following arguments to get a token:
```
spotty.exe --cache
 "I:\Portable Programs\Kodi\portable_data\userdata\addon_data\plugin.audio.spotify/spotty-cache" --ap-port 54443 --verbo
se --enable-audio-cache --name temp-spotty --get-token --scope user-read-playback-state,user-read-currently-playing,user
-modify-playback-state,playlist-read-private,playlist-read-collaborative,playlist-modify-public,playlist-modify-private,
user-follow-modify,user-follow-read,user-library-read,user-library-modify,user-read-private,user-read-email,user-read-bi
rthdate,user-top-read -u *** -p ***
```
I've tried both my email address and username with the correct password, but still the title's error message is given. Can you tell me in more detail what this error means and why it occurs? Thanks in advance!

**Comment by michaelherger** [comment]
I'm sorry, but I'm not familiar with the Kodi implementation. Under normal circumstances you shouldn't need to give your credentials, but you should use your Spotify app to authorise spotty to access your account. Please get in touch with the Kodi extension dev to see why they're expecting credentials.

**Comment by F1reP0wer12345** [comment]
My apologies for my late response. I haven't been working on spotty for almost a month. About a day after I posted the question I tried building spotty from source to check if that would make any difference. I got the spotty.exe to work, but after some attempts I still couldn't get the username and password to work. But now that've noticed your reply, I'm even more confused then I was before. How else are you supposed to connect your account? I don't get a browser popup with a Spotify authentication, And I couldn't find any arguments about using my client ID and client secret. I just assumed the --username and --password option was the only way of logging in. Are there other ways to authenticate? And if so which one do you use?

**Comment by michaelherger** [comment]
You should probably look at the LMS Plugin implementation: it fires up the helper with parameters to listen for an app to select it. And upon connection it would spit out the auth token with no need to provide credentials.

**Comment by F1reP0wer12345** [comment]
I should have probably mentioned this earlier, but when I said I compiled spotty, that was actually my first rust project. In fact I have only done 10 projects in java to begin with. Can you explain to me in simple terms what you mean by LMS Plugin? Also when I currently try to compile spotty version 1.3.0, it looks for a file called client_id.txt in the src directory. I just assumed that was the Spotify client_id right? But in order to use Spotify you would also need to get the client secret somehow. How does it work without knowing the client_secret?

**Comment by F1reP0wer12345** [comment]
I think I found the issue. I compiled it again without the --release option (I honestly don't know where I read to use the --release option), and a more detailed error shows up:
`Failed to create session: AuthenticationError(LoginFailed(PremiumAccountRequired))`
I didn't know a premium account was required. Looking back at the readme file I found this line:
_`librespot only works with Spotify Premium. This will remain the case. We will not support any features to make librespot compatible with free accounts, such as limited skips and adverts.`_
After reading this again and again, I can understand that making the application completely free without ads is not on your todo list, and making it work with Spotify's ads is even more of a hassle. Correct me if I'm interpreting it wrong.

If I want to listen to Spotify with ads, I guess I will just have to use the web version of Spotify. My goal was to turn an old computer into a Media Player, including YouTube, DVDs, Blu-rays, and if possible Spotify. I guess that last part isn't a possibility, without a subscription. I chose Kodi because I wanted full control over it using any input method I liked, Preferably without carrying an entire mouse and keyboard to the couch. It's much easier to use a remote or a controller, and both are supported.

Anyway, thanks for taking the time to reply to my questions. And who knows maybe once I do have a subscription, I might come back to adding Spotty to my project. I'll open a new issue if I face more problems at that point.
